pkg/runtime/core/value.go

```diff
@@ -18,6 +18,17 @@ type (
 		Copy() Value
 	}
 
+	// Iterable represents an interface of a value that can be iterated by using an iterator.
+	Iterable interface {
```

Comment:
`Cloneable` able to clone itself, `Measurable` able to measure itself. So `Iterable` able to iterate itself. Seems logical.

Reply:
``collections.KeyedCollection`` represents key based collection (Object) where ``collections.IndexedCollection`` represents index based collections (Array).
Iteration logic is different.